### PR TITLE
[WIP] Adding missing linalg ops wrappers

### PR DESCRIPTION
**Pull request by alexanderst**

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/Logdet.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public Logdet(SameDiff sameDiff, SDVariable input) {
+        super(sameDiff, new SDVariable[]{input});
+    }
+
+    @Override
+    public String opName() {
+        return "logdet";
+    }
+}
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/Logdet.java:52]: Need to add `calculateOutputDataTypes` method

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Qr.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    public Qr(SameDiff sameDiff, SDVariable input, boolean fullMatrices) {
+        super(sameDiff, new SDVariable[]{input});
+        addBArgument(fullMatrices);
+    }
+
+    @Override
+    public String opName() {
+        return "qr";
+    }
```
Comment (AlexDBlack) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Qr.java:48]: Needs `calculateOutputDataTypes` method

**Review by AlexDBlack — CHANGES_REQUESTED** [review]
Couple of minor things, otherwise looks fine so far.
Otherwise:
(a) looks like we need a Mmul(INDArray, INDArray) constructor added for ND4J linalg namespace,
(b) Generate and add the linalg namespace classes (using generate.sh script -> commit them here)
(c) Add Nd4j.linalg/linalg() and SameDiff.linalg/linalg() at the top of these classes
(d) Some basic tests (sanity checks) for a few of the generated SameDiff methods

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Qr.java (hunk 2 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Qr.java:48]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/Logdet.java (hunk 1 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/Logdet.java:52]: Fixed.

**Comment by AlexDBlack** [comment]
Closing in favor if this PR (which includes the same commits)
https://github.com/KonduitAI/deeplearning4j/pull/303

**Pull request closed without merging**